### ISSUE TITLE
exp and log should return dimensionless quantity

### ISSUE THREAD

**Issue by gerritholl** [hgrecco/pint#482]
`exp` and `log` rightly fail with `DimensionalityError` on any quantity that has a dimension.

But when it`s passed a dimensionless quantity, it does not return a dimensionless quantity; it returns a standard numpy array (or scalar):

```
In [85]: type(numpy.exp(ureg.Quantity(1, ureg.dimensionless)))
Out[85]: numpy.float64
```

This is inconsistent with the behaviour for other ufuncs, that always return pint quantities when passed a pint quantity.  For consistency, `log`, `exp`, and perhaps some others that I've missed, should (upon success) always return a dimensionless quantity as well.

**Comment by hgrecco** [comment]
Agreed. Would you like to submit ad PR?

**Comment by hgrecco** [comment]
Closed by c4b9d10a3b18d47248362f32522822193a6ea3fa
